source/guide_wkd.rst

```diff
@@ -0,0 +1,197 @@
+.. highlight:: console
+
+.. author:: Starkstromkonsument <it@starkstromkonsument.de>
+
+.. tag:: audience-admins
+.. tag:: mail
+.. tag:: self-hosting
+
+#######################
+Web Key Directory (WKD)
+#######################
+
+.. tag_list::
+
+[WKD] stands for "`Web Key Directory`_" and is a standard for making a users GnuPG / OpenPGP public key available via their e-mail provider or server with the domain that corresponds to their e-mail address. There's several clients (such as Enigmail in Thunderbird or OpenKeyChain on Android) that will use this standard to automatically fetch a user's public key, when writing an e-mail to them.
+
+Web Key Directories provide an easy way to discover public keys through HTTPS. They provide an important piece to the infrastructure to improve the user experience for exchanging secure emails and files. In contrast to the public keyservers a Web Key Directory does not publish mail addresses and it is an authoritative pubkey source for its domain.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * e-mail encryption with GnuPG / OpenPGP 
+  * :manual:`domains <web-domains>`
+
+Prerequisites
+=============
+
+You need the public keys for the e-mail identities you want to provide via WKD (e.g. isabell@stardust.uberspace.de). This guide does not explain how to generate GnuPG / OpenPGP keys.
+
+The domain for the advanced method has to be set up at your DNS-Hoster and within your uberspace:
+
+::
+
+ [isabell@stardust ~]$ uberspace web domain list
```

Comment:
Please use the snippet like so `.. include:: includes/web-domain-list.rst`.

Reply:
I did not use the snippet because I wanted to point out that the subdomain openpgpkey.[...] has to exist. Do you want me to change it anyway?

Reply:
Judging by the official docs the `openpgpkey.` subdomains seem to be more of a workaround and are not recommended. Can we get by by using only `.well-known/`?

Reply:
Hi luto, can you provide a link to the docs you refer to please? In case you refer to the stopgap-method described at https://wiki.gnupg.org/WKD, were are not doing that in the proposed guide. We are using our own server (or webspace ;-) and do not redirect to a third party.

IMHO the openpgpkey-subdomain (Advanced method) is queried first. Only if it does not resolve by DNS, the direct method is used as a fallback (as pointed out here: https://wiki.gnupg.org/WKDHosting#Publishing). This could be a problem for people who have Wildcard-DNS-Entries set up.

Another guide at https://keys.openpgp.org/about/usage#wkd-as-a-service only describes the advanced method and does not even mention the direct one.

I would like to leave it in the guide to be fully compliant.

I propose to rework my guide to use those examples:

- isabell@example.org
- https://openpgpkey.example.org/.well-known/openpgpkey/example.org/hu/
- https://example.org/.well-known/openpgpkey/hu/

Reply:
@Marmo What do you think about this, as the one who initially opened the issue?

Reply:
I managed to miss-read the docs then, sorry! :/ the proposed trio of placeholders are fine. Thanks!

Reply:
Thank you for your feedback. I will push a commit, changing the Domain to example.org. What about the snippet `includes/web-domain-list.rst`? Should I use it anyway or not?

Reply:
no snippet in that case. Maybe a hint that the entire guide isn't very useful, if you don't own a domain.